### PR TITLE
Unit tests for EventDispatcher

### PR DESCRIPTION
**Pull request by squall0gd**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by squall0gd** [comment]
Current coverage ~89%

File: pkg/kubelet/cm/event_dispatcher.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher.go]: :) please remove

**Review by flyingcougar — CHANGES_REQUESTED** [review]

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: add check for socket/name correlation corectness

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: here is inside for loop, in test case above outside, please be consistent

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please also check for event

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: s/dummy/`sth else`

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please rename ResetDummyObject

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please add comment on that case

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please add comment on that case

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please add comment on that case

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: please rename to sth else, more descriptive

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: to consider: instead doing `switch case` here, adjust the `testCases` to consist of output from isolators and expected `ResourceConfig`. the in the loop you just need to compare those. The advantage is that adding new test cases involves only appending to `testCase` structure.

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (flyingcougar) [inline pkg/kubelet/cm/event_dispatcher_test.go]: i.e. s/testCase/testResourceConfig
sth more descriptive

**Review by flyingcougar — CHANGES_REQUESTED** [review]

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: Disagree. Let's be consistent with Kubernetes

File: pkg/kubelet/cm/event_dispatcher_test.go
Comment (squall0gd) [reply pkg/kubelet/cm/event_dispatcher_test.go]: done

**Comment by squall0gd** [comment]
Current coverage > 90%

**Comment by flyingcougar** [comment]
lgtm from me, please check if tests with master branch.

**Review by ConnorDoyle — COMMENTED** [review]
Please squash to 1 commit before merging

**Pull request merged**
